### ISSUE TITLE
[Bug] Coroutine cancels 

### ISSUE THREAD

**Issue by polyzos** [NephyProject/Penicillin#19]
Hi im using the library to retrieve all followers of some users based on their id, using the listIdsByScreenName() function (tried other functions as well but with the same outcome).
When using these functions i get the following exception at random every time
```
Exception in thread "main" kotlinx.coroutines.JobCancellationException: Parent job is Cancelling; job=CompletableDeferredImpl{Cancelling}@7d3815f7
Caused by: kotlinx.coroutines.channels.ClosedReceiveChannelException: Channel was closed
	at kotlinx.coroutines.channels.Closed.getReceiveException(AbstractChannel.kt:1081)
	at kotlinx.coroutines.channels.AbstractChannel$ReceiveElement.resumeReceiveClosed(AbstractChannel.kt:913)
	at kotlinx.coroutines.channels.AbstractSendChannel.helpClose(AbstractChannel.kt:323)
	at kotlinx.coroutines.channels.AbstractSendChannel.close(AbstractChannel.kt:262)
	at kotlinx.coroutines.channels.ChannelCoroutine.close(ChannelCoroutine.kt)
	at kotlinx.coroutines.channels.SendChannel$DefaultImpls.close$default(Channel.kt:94)
	at io.ktor.network.tls.TLSClientHandshake$input$1.invokeSuspend(TLSClientHandshake.kt:92)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:233)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:742)

```
Any ideas why the job might be cancelling by itself at random, without fetching all the followers first?
Best Regards

**Comment by SlashNephy** [comment]
Hi polyzos. Please let me know the following information.

- Environment (OS, Java version, Kotlin version, Ktor version, Penicillin library version)
- Bug reproduction code
- The Ktor Client you are using (CIO or Apache or ...)

Thanks

**Comment by polyzos** [comment]
I'm running on linux with java8 and kotlin 1.3.20
This is my code that fetching user's followers for around 250 user screen_names
```
  val ids =
                client.followers.listIdsByScreenName(screenName)
                    .untilLast().fold(emptySequence<Long>()) { acc, nextRes ->
                        if (nextRes.rateLimit.remaining == 0) {
                            println("Sleeping for 15 minutes because rate limit exceeded")
                            Thread.sleep(60 * 15 * 1000 + 500)
                        }
                        println(nextRes.result.ids.size)
                        acc + nextRes.result.ids
                    }.toList()
```

and my build.gradle file
```
sourceCompatibility = 1.8

repositories {
    mavenCentral()
    jcenter()
    maven { url "https://kotlin.bintray.com/ktor" }
    maven { url "https://kotlin.bintray.com/kotlinx" }
    maven { url "https://kotlin.bintray.com/kotlin-eap" }
    maven { url "https://dl.bintray.com/nephyproject/penicillin" }
}


dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
    testCompile group: 'junit', name: 'junit', version: '4.12'
    
    // Spark
    compile group: 'org.apache.spark', name: 'spark-core_2.12', version: '2.4.0'
    compile group: 'org.apache.spark', name: 'spark-sql_2.12', version: '2.4.0'

    compile group: 'com.natpryce', name: 'konfig', version: '1.6.10.0'

    implementation("org.neo4j:neo4j-ogm-core:3.0.2") {
        exclude  group: 'com.fasterxml.jackson.core'
    }

    implementation("org.neo4j:neo4j-ogm-bolt-driver:3.0.2") {
        exclude  group: 'com.fasterxml.jackson.core'
    }

    implementation "jp.nephy:penicillin:4.1.1"
    implementation "io.ktor:ktor-client-cio:1.1.3"
}

compileKotlin {
    kotlinOptions.jvmTarget = "1.8"
}
compileTestKotlin {
    kotlinOptions.jvmTarget = "1.8"
}
```

**Comment by SlashNephy** [comment]
In https://github.com/NephyProject/Penicillin/commit/e6b72f9cea70129fd526b0c08c14b1e14978f4f1, CursorJsonObjectApiAction<M>#untilLast() began to consider rate limits.  
So you no longer need to run `Thread.sleep` manually.

```kotlin
val ids: List<Long> = client.followers.listIdsByScreenName(screenName, count = 5000).untilLast().allIds
```

`/followers/ids` [endpoint](https://developer.twitter.com/en/docs/accounts-and-users/follow-search-get-users/api-reference/get-followers-ids) accepts count up to 5000. I recommend `count = 5000` parameter.  

This EAP build is available as `4.1.3-eap-2`.

By the way, `JobCancellationException` may be caused by a mismatch between the Ktor version used in Penicillin and the Ktor version defined in build.gradle.  
Please try upgrading Penicillin and Ktor in build.gradle.  

```gradle
dependencies {
    // ...

    implementation "jp.nephy:penicillin:4.1.3-eap-2"
    implementation "io.ktor:ktor-client-cio:1.1.3"
}
```

If it does not solve the problem, try replacing `io.ktor:ktor-client-cio:1.1.3` with `io.ktor:ktor-client-apache:1.1.3` in build.gradle.  
CIO seems to be unstable in some environments.
